dcmdata/libsrc/dcchrstr.cc

```diff
@@ -139,16 +139,194 @@ OFCondition DcmCharString::verify(const OFBool autocorrect)
     return errorFlag;
 }
 
+unsigned long DcmCharString::getVM()
+{
+    unsigned long vm;
+    const char* start = NULL;
+    const char* end = NULL;
+    if (getIndexOfPosition(-1, start, end, vm).good())
+    {
+        return vm;
+    }
+    return 0;
+}
+
+
+OFCondition DcmCharString::getOFString(OFString& stringVal, const unsigned long pos, OFBool /*normalize*/)
+{
+    unsigned long vm;
+    const char* start;
+    const char* end;
+    OFCondition result = getIndexOfPosition(pos, start, end, vm);
+    if (result.bad())
+        return result;
+
+    if (vm == 0 || start == end) {
+        stringVal.clear();
+        return EC_Normal;
+    }
 
-OFBool DcmCharString::containsExtendedCharacters(const OFBool /*checkAllStrings*/)
+    stringVal.assign(start, end);
+    return EC_Normal;
+}
+
+
+OFCondition DcmCharString::getIndexOfPosition(const long pos, const char*& start, const char*& end, unsigned long& vm)
 {
-    OFBool result = OFFalse;
+    // the vast majority of values have VM 0 or 1, so optimize for these
```

Comment:
I'm not too happy with this helper method, but I also didn't want to copy the relevant code into both getVM() and getOFString().
I'm also not sure about the performance impact. My assumption is that the actual character set checks are only needed for very few tags (multi-valued tags with a single-value multi-byte encodings or with code extensions).